src/Blueprint/Blueprint/Models/PSBlueprintAssignment.cs

```diff
@@ -59,7 +60,12 @@ internal static PSBlueprintAssignment FromAssignment(Assignment assignment)
                 BlueprintId = assignment.BlueprintId,
                 ProvisioningState = PSAssignmentProvisioningState.Unknown,
                 Status = new PSAssignmentStatus(),
-                Locks = new PSAssignmentLockSettings {Mode = PSLockMode.None},
+                Locks = new PSAssignmentLockSettings
+                {
+                    Mode = PSLockMode.None,
+                    ExcludedActions = assignment.Locks.ExcludedActions.ToList(),
+                    ExcludedPrincipals = assignment.Locks.ExcludedPrincipals.ToList()
```

Comment:
>assignment.Locks.ExcludedPrincipals.ToList() [](start = 41, length = 44)

What happens when ExcludedPrincipals is null?

Reply:
Just realized that.